docs/next/guides/upgrade-and-migration/migrating-from-11.1-to-12.0.md

```diff
@@ -12,3 +12,225 @@ pageClass: migration-guide
+
+#### After
+
+A plugin's [`updatePlugin()`](@/api/autocolumnsize.md#updateplugin) method gets triggered only when the object passed to [`updateSettings()`](@/api/core.md#updatesettings) contains at least one of the following:
+- The plugin's [`PLUGIN_KEY`](@/guides/building-and-testing/plugins.md#_2-extend-the-baseplugin) (the plugin's main alias)
+- An entry from the plugin's [`SETTING_KEYS`](@/api/baseplugin.md#setting-keys) (a property that stores all additional settings relevant to the plugin)
+
+As a result, a plugin gets updated only if you update settings related to that particular plugin.
+
+##### Built-in plugins
+
+The change above affects most of Handsontable's [built-in plugins](@/api/plugins.md) as well.
+
+Starting with Handsontable 12.0.0, a built-in plugin gets updated on a call to [`updateSettings()`](@/api/core.md#updatesettings) only if the passed configuration object contains an option relevant to that particular plugin. For example, the [`Autofill`](@/api/autofill.md) plugin gets updated if your call to [`updateSettings()`](@/api/core.md#updatesettings) updates the [`fillHandle`](@/api/options.md#fillhandle) option.
+
+To find out which options a given plugin observes, see that plugin's [`updatePlugin()` API reference](@/api/autofill.md#updateplugin).
+
+::: tip
+The following built-in plugins still get updated on every [`updateSettings()`](@/api/core.md#updatesettings) call:
+  - [`AutoColumnSize`](@/api/autocolumnsize.md)
+  - [`AutoRowSize`](@/api/autorowsize.md)
+  - `TouchScroll`
+  - [`UndoRedo`](@/api/undoredo.md)
+:::
+
+#### Migrating to Handsontable 12.0
+
+If you want your [custom plugin](@/guides/building-and-testing/plugins.md) to still get updated on every [`updateSettings()`](@/api/core.md#updatesettings) call, set your plugin's [`SETTING_KEYS`](@/api/baseplugin.md#setting-keys) to `true`:
+
+```js
+static get SETTING_KEYS() {
+  return true;
+}
+```
+
+However, in most cases, it's better to provide an explicit list of configuration options that your custom plugin observes. For details, see the [Plugins](@/guides/building-and-testing/plugins.md) guide.
+
+## Step 3: Adjust to the `afterDocumentKeyDown` changes
+
+Handsontable 12.0.0 changes how the [`afterDocumentKeyDown`](@/api/hooks.md#afterdocumentkeydown) hook works.
+
+#### Before
+
+When you used a keyboard to navigate an internal Handsontable instance (e.g., a context menu), the [`afterDocumentKeyDown`](@/api/hooks.md#afterdocumentkeydown) hook got fired on every key press.
+
+```js
+afterDocumentKeyDown() {
+  // when you navigate the grid, the console message gets logged on every key press
+  // when you navigate the context menu, the console message also gets logged on every key press
+  console.log('key pressed');
+}
+```
+
+#### After
+
+When you use a keyboard to navigate some internal instances of Handsontable (e.g., a context menu) hook may not get fired anymore.
```

Comment:
The name of the `afterDocumentKeyDown` hook was removed from here yesterday. Is this on purpose?

Reply:
No, fixed: [2e5f69c](https://github.com/handsontable/handsontable/pull/9271/commits/2e5f69cbdaa2c146e95552c82dc723e3c9657a8d)